rfaas/include/rfaas/executor.hpp

```diff
@@ -80,13 +94,55 @@ namespace rfaas {
 
     // Skipping managers is useful for benchmarking
     bool allocate(std::string functions_path, int numcores, int max_input_size, int hot_timeout,
-        bool skip_manager = false, rdmalib::Benchmarker<5> * benchmarker = nullptr);
-    void deallocate();
-    rdmalib::Buffer<char> load_library(std::string path);
+        bool skip_manager = false, rdmalib::Benchmarker<5> * benchmarker = nullptr)
+    {
+      return static_cast<Derived*>(this)->allocate(functions_path, numcores, max_input_size, hot_timeout, skip_manager, benchmarker);
+    }
+    void deallocate()
+    {
+      static_cast<Derived*>(this)->deallocate();
+    }
+    rdmalib::Buffer<char, Library> load_library(std::string path)
+    {
+      return static_cast<Derived*>(this)->load_library(path);
+    }
+    void poll_queue()
+    {
+      static_cast<Derived*>(this)->poll_queue();
+    }
+
+    template<typename T, typename U>
+    std::future<int> async(std::string fname, const rdmalib::Buffer<T, Library> & in, rdmalib::Buffer<U, Library> & out, int64_t size = -1);
+
+    template<typename T,typename U>
+    std::future<int> async(std::string fname, const std::vector<rdmalib::Buffer<T, Library>> & in, std::vector<rdmalib::Buffer<U, Library>> & out);
+    bool block();
+
+    // FIXME: irange for cores
+    // FIXME: now only operates on buffers
+    //template<class... Args>
+    //void execute(int numcores, std::string fname, Args &&... args)
+    template<typename T, typename U>
+    std::tuple<bool, int> execute(std::string fname, const rdmalib::Buffer<T, Library> & in, rdmalib::Buffer<U, Library> & out);
+
+    template<typename T>
+    bool execute(std::string fname, const std::vector<rdmalib::Buffer<T, Library>> & in, std::vector<rdmalib::Buffer<T, Library>> & out);
+  };
+
+  struct libfabric_executor : executor<libfabric_executor, libfabric> {
```

Comment:
Why are separate implementations for ibverbs and libfabric necessary? This is the executor from `rfaas` - shouldn't they just use the concepts from `rdmalib`? Why the need to duplicate executors?